packages/py/polywrap-msgpack/tests/test_msgpack.py

```diff
@@ -71,3 +69,126 @@ def test_generic_map_decode():
+
+# STRINGS
+
+def test_sanitize_str_returns_same_str():
+    assert sanitize("https://docs.polywrap.io/") == "https://docs.polywrap.io/"
+
+#def test_sanitized_polywrap_ens_uri(UriPath):
+#    pass
+
+# LISTS
+
+def test_sanitize_simple_list_returns_simple_list():
+    assert  sanitize([1]) == [1]
+
+def test_sanitize_empty_list_returns_empty_list():
+    assert sanitize([]) == []
+
+def test_sanitize_long_list_returns_long_list():
+    assert sanitize([2,55,1234,6345]) == [2,55,1234,6345] 
+
+def test_sanitize_complex_list_returns_list(complex_list):
+    assert sanitize(complex_list) == complex_list
+
+def test_sanitize_nested_list_returns_nested_list(nested_list):
+    assert sanitize(nested_list) == nested_list 
+
+# COMPLEX NUMBERS
+
+def test_sanitize_complex_number_returns_string():
+    assert sanitize(3 + 5j) == "(3+5j)"
+    assert sanitize(0 + 9j) == "9j"
+
+def test_sanitize_simple_dict_returns_sanitized_values(simple_dict):
+    assert sanitize(simple_dict) == simple_dict
+
+# SLOTS
+
+def test_sanitize_object_with_slots_attributes_returns_dict_instead(object_with_slots_attributes, object_with_slots_sanitized):
+    assert sanitize(object_with_slots_attributes) == object_with_slots_sanitized
+
+# TUPLES
+
+def test_sanitize_single_tuple_returns_list(single_tuple):
+    # To create a tuple with only one item, you have add a comma after the item, 
+    # otherwise Python will not recognize the variable as a tuple.
+    assert type(sanitize(single_tuple)) == list
+    assert sanitize(single_tuple) == [8] 
+
+def test_sanitize_long_tuple_returns_list():
+    assert sanitize((2,3,6)) == [2,3,6] 
+
+def test_sanitize_nested_tuples_returns_nested_list(nested_tuple, nested_list):
+    assert sanitize(nested_tuple) == nested_list
+
+
+# SETS
+
+def test_sanitize_set_returns_list(set1):
+    # Remember sets automatically reorganize the contents of the object
+    # meaning {'bob', 'alice'} might be stored as {'alice','bob'} in memory
+    assert type(sanitize(set1)) == list
+
+def test_sanitize_set_returns_list_with_all_items_of_the_set(set1, set2):
+    sanitized = sanitize(set1)
+    r = []       
+    [r.append(True) if item in sanitized else r.append(False) for item in set1]
+    assert False not in r
+    
+    sanitized = sanitize(set2)
+    r = []       
+    [r.append(True) if item in sanitized else r.append(False) for item in set2]
+    assert False not in r
+
+def test_sanitize_set_returns_list_of_same_length(set1):
+    assert len(sanitize(set1)) == len(set1)
+
+
+def test_sanitize_complex_dict_returns_sanitized_values():
+    # This test is not passing because when we sanitize a set, the contents are shuffled around.
+    complex_dict = {'name': ['John', 'Doe'],
+        'position':[-0.34478,12.98453],
+        'color': 'green',
+        'age':33,
+        'origin':(0,0),
+        'is_online': True,
+        'pet': None,
+        'friends': {'bob','alice','megan','john'} }
```

Comment:
is the attribute `friends` here expected to be a dict or a list? I think it should be changed to a list, right?

Reply:
It's input as a python set (https://www.w3schools.com/python/python_sets.asp), and sets once sanitized are expected to output a lists ([see line 166](https://github.com/polywrap/toolchain/pull/1273/files/a5ba60c83d4b9e45aa14494e50a81a721da9fd17#diff-ce8abd84375041768c63587bc34514c95ac1797881274ef52f433a62b65e4ebfR166))

this though makes the pytest fail at times because the sets dont have an specific order, so when we sanitize them into a list they are scrambled into a newly-ordered list which doesnt always have its items in the same order as we wrote them on the set on Line 158. thus failing every now and then when we `assert sanitize(complex_dict) == sanitized_complex_dict
`

Reply:
Gotcha! What about converting it into an ordered list and comparing it that way? Hence, guaranteeing that tests wont fail